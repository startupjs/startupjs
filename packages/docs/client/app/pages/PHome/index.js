import { useEffect } from 'react'
import { observer, emit, useLocal } from 'startupjs'
import { useDocsContext } from '../../../../docsContext'
import { useLang } from '../../../clientHelpers'
import { DEFAULT_LANGUAGE } from '../../../const'

export default observer(function PHome ({
  style
}) {
  const docs = useDocsContext()
  const [params] = useLocal('$render.params')
  const paramsLang = params.lang
  let [lang = DEFAULT_LANGUAGE, setLang] = useLang()
  if (paramsLang && paramsLang !== lang) {
    lang = paramsLang
    setLang(lang)
  }

  function getDocPath (docs) {
    let path = '/'
    const docName = Object.keys(docs)[0]
    const doc = docs[docName]
    switch (doc.type) {
      case 'mdx':
        path += docName
        break
      case 'collapse':
        path += docName
        if (!doc.component) path += getDocPath(doc.items)
        break
    }
    return path
  }

  useEffect(() => {
    emit(
      'url',
      `/docs/${lang}` + getDocPath(docs),
      { replace: true }
    )
  }, [])
  return null
})
