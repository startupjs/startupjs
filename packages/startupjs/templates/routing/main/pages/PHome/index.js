import React from 'react'
import { observer } from 'startupjs'
import { ScrollView } from 'react-native'
import { TestComponent } from 'components'
import './index.styl'

export default observer(function PHome () {
  return pug`
    ScrollView.root
      TestComponent
  `
})
