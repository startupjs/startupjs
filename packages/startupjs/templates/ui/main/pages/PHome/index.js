import React from 'react'
import { observer } from 'startupjs'
import { ScrollView } from 'react-native'
import { TestComponent } from 'components'
import './index.styl'
import { Content } from '@startupjs/ui'

export default observer(function PHome () {
  return pug`
    ScrollView.root
      Content
        TestComponent
  `
})
