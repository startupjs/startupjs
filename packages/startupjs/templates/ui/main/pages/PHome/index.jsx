import React from 'react'
import { ScrollView } from 'react-native'
import { observer } from 'startupjs'
import { Content } from '@startupjs/ui'
import { TestComponent } from 'components'
import './index.styl'

export default observer(function PHome () {
  return pug`
    ScrollView.root
      Content
        TestComponent
  `
})
