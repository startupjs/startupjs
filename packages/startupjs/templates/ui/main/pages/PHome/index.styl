.root
  padding 2u 0
  flex 1
