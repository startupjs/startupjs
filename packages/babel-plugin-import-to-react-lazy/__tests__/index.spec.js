const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index')
const { name: pluginName } = require('../package.json')

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {},
  tests: {
    'Ignores files without @asyncImports annotation': /* js */`
      console.log('Hello World')
    `,
    'Processes files with @asyncImports annotation and imports lazy automatically': /* js */`
      /* @asyncImports */
      console.log('Hello World')
    `,
    'Imports lazy and transforms import to use it': /* js */`
      /* @asyncImports */
      import PHome from './PHome'
      import PGames from './PGames'
      export { PHome, PGames }
      console.log('Hello World')
    `,
    'Imports lazy and transforms export to use it': /* js */`
      /* @asyncImports */
      export { default as PHome } from './PHome'
      export { default as PGames } from './PGames'
      console.log('Hello World')
    `
  }
})
