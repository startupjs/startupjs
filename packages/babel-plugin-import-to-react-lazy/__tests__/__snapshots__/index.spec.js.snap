// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-import-to-react-lazy Ignores files without @asyncImports annotation: Ignores files without @asyncImports annotation 1`] = `

console.log('Hello World')

      ↓ ↓ ↓ ↓ ↓ ↓

console.log("Hello World");


`;

exports[`@startupjs/babel-plugin-import-to-react-lazy Imports lazy and transforms export to use it: Imports lazy and transforms export to use it 1`] = `

/* @asyncImports */
export { default as PHome } from './PHome'
export { default as PGames } from './PGames'
console.log('Hello World')

      ↓ ↓ ↓ ↓ ↓ ↓

import { lazy as _lazy } from "react";

/* @asyncImports */
export const PHome = _lazy(() => import("./PHome"));
export const PGames = _lazy(() => import("./PGames"));
console.log("Hello World");


`;

exports[`@startupjs/babel-plugin-import-to-react-lazy Imports lazy and transforms import to use it: Imports lazy and transforms import to use it 1`] = `

/* @asyncImports */
import PHome from './PHome'
import PGames from './PGames'
export { PHome, PGames }
console.log('Hello World')

      ↓ ↓ ↓ ↓ ↓ ↓

import { lazy as _lazy } from "react";

/* @asyncImports */
const PHome = _lazy(() => import("./PHome"));

const PGames = _lazy(() => import("./PGames"));

export { PHome, PGames };
console.log("Hello World");


`;

exports[`@startupjs/babel-plugin-import-to-react-lazy Processes files with @asyncImports annotation and imports lazy automatically: Processes files with @asyncImports annotation and imports lazy automatically 1`] = `

/* @asyncImports */
console.log('Hello World')

      ↓ ↓ ↓ ↓ ↓ ↓

import { lazy as _lazy } from "react";

/* @asyncImports */
console.log("Hello World");


`;
